torchvision/prototype/transforms/functional/_geometry.py

```diff
@@ -870,6 +870,16 @@ def elastic(
 
 
 elastic_transform = elastic
+# datumbox: we should decide upon the naming of those that differ from the old API. Others like this include vflip,
+# hflip, pil_to_tensor, to_pil_image. Some of them might not be worth renaming to avoid changing user code; those that we do need to change them
+# we should offer aliases that throw a deprecation warning. First we need to get a list of those functionals that
+# were renamed.
+
+# datumbox: Some kernels don't have mid-level dispachers. Examples:
+# get_dimensions, five_crop, ten_crop
+# Also some others don't have their names exposed via the prototype functional at all. Here are some of them:
+# to_tensor, get_image_size, get_image_num_channels, convert_image_dtype
+# We should check which of them should be deprecated and which should be exposed normally via an alias
```

Comment:
cc @vfdev-5 / @pmeier

Reply:
I'll do it.

Reply:
Can't we have both alias and another name without deprecation, just an alias ?
For example,  `hflip` and `horizontal_flip` could work.

Reply:
We could. But do we really have to change it? `hflip` is extremely prevalent, why rename? And yes we could offer an alias but then we are pushing for 2 idioms at the same time. I personally would prefer to use our "BC-breaking" and "deprecation" budget wisely to make the API consistent but not make unnecessary changes that work. That's one opinion though.

Reply:
See https://github.com/pytorch/vision/pull/6492/commits/024f2f450bc2aa69c7a21983432cd05c685ff603 for `five_crop` and `ten_crop`

Reply:
My reasoning for `hflip` -> `horinzontal_flip` was that currently the name of the functional and the name of the transform diverge:

https://github.com/pytorch/vision/blob/0eb8aabd9fd5950d3867dc8e6b6b81006c8d7416/torchvision/transforms/transforms.py#L681

I would prefer to keep the names aligned. To keep BC, I would follow Victors suggestion and simply do

```py
hflip = horizontal_flip
```

Reply:
This is the full list:

- `convert_image_dtype`
- `five_crop`
- `get_dimensions`
- `get_image_num_channels`
- `get_image_size`
- `hflip`
- `pil_to_tensor`
- `ten_crop`
- `to_pil_image`
- `to_tensor`
- `vflip`

Reply:
Kernels that you already added on #6492:
- five_crop
- ten_crop

Kernels for which we need aliases (no deprecation warning):
- hflip (from `horizontal_flip`)
- vflip (from `vertical_flip`)
- to_pil_image (from `to_image_pil`)
- pil_to_tensor (from `to_image_tensor`)

Missing kernels we need to create / expose on the new API:
- convert_image_dtype
- get_dimensions
- get_image_num_channels (new tmp name `get_channels_dim()` + alias for the old)
- get_image_size (new tmp name `get_spatial_dims()` + alias for the of)
- to_tensor (with deprecation warning to use `convert_image_dtype()` + `to_image_tensor()`)

I also noticed that in many places in `prototype.transforms` we still import from `torchvision.transforms.functional`. We shouldn't do this. We should be instead importing from `prototype.transforms.functional` instead. I know that the majority of them will be alias or imported from the stable `F` but not doing so is a good test that proves we offer all the kernels we need.

Reply:
#6500